Implement traits for `IndexMap` and `IndexSet`. (See <https://docs.rs/indexmap>.

tage64 opened the pull request:
…/indexmap>)

tage64 committed:
Implement traits for `IndexMap` and `IndexSet`. (See <https://docs.rs/indexmap>)

tage64 committed:
Remove some unnecessary bounds and add some more trait implementations to indexmap.

timothee-haudebourg commented on src/impls/indexmap/mod.rs:
```diff
@@ -0,0 +1,2 @@
+mod indexmap;
```
I'm getting a `clippy` warning here:
```
warning: module has the same name as its containing module
 --> src/impls/indexmap/mod.rs:1:1
  |
1 | mod indexmap;
  | ^^^^^^^^^^^^^
  |
  = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#module_inception
  = note: `#[warn(clippy::module_inception)]` on by default
```

timothee-haudebourg reviewed (changes requested):
Thanks for the addition, there is just one warning to fix before I can merge.

Commit:
Add CI.

Commit:
Add `all-impl` feature. Fix CI.

tage64 committed:
Move indexmap and indexset to there own modules and remove the indexmap-module that was previously their parent.

tage64 commented:
Hopes that last commit solves the problem. I however didn't get that warning when I was just running `cargo clippy`. Perhaps I should specify some arguments to clippy, but then those should probably be stored in a `Clippy.toml`-file or something like that.

timothee-haudebourg commented:
> I however didn't get that warning when I was just running `cargo clippy`.

I think that's a version problem. Every release new warnings are added to `clippy`, do you have the latest? Also can you rebase your contribution on top of the `master` branch so it includes the CI I added.

timothee-haudebourg commented on src/impls/mod.rs:
```diff
@@ -12,3 +12,8 @@ mod serde_json;
+#[cfg(feature = "indexmap")]
+mod indexmap;
+#[cfg(feature = "indexmap")]
+mod indexset;
```
I would really prefer to have each external crate implemented in a single separate module, just like you did at first. You can solve the clippy warning either by renaming the submodules in `indexmap` `map` and `set`, or by adding `#[allow(clippy::module_inception)]` (I think it can be a justified use of module inception).

tage64 committed:
Revert "Move indexmap and indexset to there own modules and remove the indexmap-module that was previously their parent."

This reverts commit cff331429fea91d5b89373e7ea94d2173610ff7f.

tage64 committed:
Rename indexmap::indexmap and indexmap::indexset to indexmap::map and indexmap::set to mittigate clippy::module_inception.

tage64 commented:
I moved indexmap and indexset to indexmap::map and indexmap::set modules.

jeandudey committed:
Add no_std support. (#7)

- Renamed the "nostd" feature to "std" in order to follow the convention
  with most crates and to avoid any crate specifying the "nostd" feature
  as another crate might want "std" collections but it was disabled by the
  other crate.

- Added an "alloc" feature in order to support no_std but additionally
  alloc structures.

- Separated implementation modules for both "std" and "alloc".

Testing can be done as:

```
cargo build --no-default-features

cargo build --no-default-features --features "alloc"

cargo build
```

Signed-off-by: Jean-Pierre De Jesus DIAZ <me@jeandudey.tech>

timothee-haudebourg commented:
Unfortunately you'll have to get up to date with the `master` branch once again since #7.

timothee-haudebourg commented on Cargo.toml:
```diff
@@ -12,11 +12,15 @@ license = "MIT/Apache-2.0"
+alloc = []
+std = []
 nightly = []
+all-impls = ["slab", "smallvec", "serde_json", "ijson"]
```
Can you add the `indexmap` feature to `all-impls`?

timothee-haudebourg commented on CHANGELOG.md:
```diff
@@ -5,6 +5,16 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
```
Add your changes in this section.
